CMakeLists.txt

```diff
@@ -174,6 +174,7 @@ target_sources(scylla-main
     compress.cc
     converting_mutation_partition_applier.cc
     counters.cc
```

Comment:
> As of this patch, it works like this:
> every `$tick_period` (15 minutes), if we are the current Raft leader,
> we check for dict-aware tables which have no dict, or a dict older
> than `$retrain_period`.

Let's say we have a good dictionary. Doesn't this mean we'll sample every 15 minutes, only to throw the samples away? 

I think it's too much, for large clusters. Once a day should be enough, data doesn't change that often.

Reply:
> > As of this patch, it works like this:
> > every `$tick_period` (15 minutes), if we are the current Raft leader,
> > we check for dict-aware tables which have no dict, or a dict older
> > than `$retrain_period`.
> 
> Let's say we have a good dictionary. Doesn't this mean we'll sample every 15 minutes, only to throw the samples away?

Hmm, right, yes. It's wasteful.

If we find out that the current dictionary is already good, it probably makes sense to refresh its timestamp (the `timestamp` column, not the internal timestamp), so that the trainer doesn't retry this table again for some time, to minimize the noise. It will be a small change.

Reply:
Yes